includes/abstracts/abstract-wc-order.php

```diff
@@ -780,14 +780,36 @@ public function get_item_count( $item_type = '' ) {
 	}
 
 	/**
-	 * Get an order item object, based on it's type.
+	 * Get an order item object, based on it's ID. The item must belong to the current
+	 * order. If the item cannot be found or it doens't belong to current order
+	 * FALSE will be returned.
 	 *
 	 * @since  3.0.0
 	 * @param  int $item_id
 	 * @return WC_Order_Item
 	 */
 	public function get_item( $item_id ) {
-		return WC_Order_Factory::get_order_item( $item_id );
+		foreach ( $this->items as $items ) {
```

Comment:
> On re-review I don't like how we're using woocommerce_order_get_items filter after this. What is someone is appending more items to woocommerce_order_get_items themselves?

I don't like it either. It's a bit pointless unless someone is using this filter to hide or swap an item object.

> If this was made to use get_items instead, so all items of a type are loaded and indexed in memory before returning one, would that work better?

I can do a quick SQL query to fetch `order_item_type` and call `get_items( $type )`. It would be a bit inefficient though, because changes are we already have the item in memory already loaded. It's up to you.